lib/exception_handling.rb

```diff
@@ -456,6 +461,41 @@ def clean_backtrace(exception)
 
     private
 
+    # @param exception_info [ExceptionInfo]
+    #
+    # @return [Array<String>]
+    def tags_for_honeybadger(exception_info)
+      (
+        honeybadger_auto_tags(exception_info.exception) +
+        exception_info.honeybadger_tags +
+        honeybadger_tags_from_log_context(exception_info.honeybadger_context_data)
+      ).uniq
+    end
+
+    # @param exception [Exception]
+    #
+    # @return [Array<String>]
+    def honeybadger_auto_tags(exception)
+      @honeybadger_auto_tagger&.call(exception) || []
+    rescue => ex
+      traces = ex.backtrace.join("\n")
+      message = "Unable to execute honeybadger_auto_tags callback. #{ExceptionHandling.encode_utf8(ex.message.to_s)} #{traces}\n"
+      ExceptionHandling.log_info(message)
+      []
+    end
+
+    def honeybadger_tags_from_log_context(honeybadger_context_data)
+      if @honeybadger_log_context_tags
+        @honeybadger_log_context_tags.map do |tag_name, tag_path|
+          if (value_from_log_context = honeybadger_context_data.dig(:log_context, *tag_path))
+            "#{tag_name}:#{value_from_log_context}"
+          end
+        end.compact
```

Comment:
Instead of `map.compact`, how do you feel about `flat_map`? Just an idea.
```
@honeybadger_log_context_tags.flat_map do |tag_name, tag_path|
  if (value_from_log_context = honeybadger_context_data.dig(:log_context, *tag_path))
    ["#{tag_name}:#{value_from_log_context}"]
  else
    []
  end
```